### ISSUE TITLE
require(...) is relative to the process working directory instead of the file it is called from

### ISSUE THREAD

**Issue by nlowe** [StackExchange/dnscontrol#440]
Let's say I have this file structure

```txt
.
├── dnsconfig.js
└── k8s
    ├── clusters
    │   └── my-cluster.js
    ├── clusters.js
    └── util.js
```

With the following file contents:

```javascript
// ./dnsconfig.js
require('./k8s/clusters.js')

// ...
```

```javascript
// ./k8s/clusters.js
require('./clusters/my-cluster.js')

// ...
```

```javascript
// ./k8s/clusters/my-cluster.js
require('../util.js')

// ...
```

`dnsconfig` will fail to push the zone:

```bash
# dnscontrol push --config ./dnsconfig.js
requiring: ./k8s/clusters.js
requiring: ./clusters/my-cluster.js
Executing javascript in ./dnsconfig.js: Error: Error: open ./clusters/my-cluster.js: no such file or directory
```

**Comment by tlimoncelli** [comment]
Hmm... it seems like require() doesn't change the CWD.

Your example works if you specify all paths from the root directory:

```
==> dnsconfig.js <==
// ./dnsconfig.js
require('./k8s/clusters.js')


==> k8s/clusters.js <==
// ./k8s/clusters.js
require('./k8s/clusters/my-cluster.js')


==> k8s/util.js <==


// foo
//

==> k8s/clusters/my-cluster.js <==

// ./k8s/clusters/my-cluster.js
require('./k8s/util.js')
```

**Comment by nlowe** [comment]
Yeah I found that later this afternoon. However, this isn't how `require(...)` behaves in node, which is why I was initially confused. Is it something you'd accept a PR for? Or can it at least be documented better? I was only able to find this feature by searching for it on github.

**Comment by tlimoncelli** [comment]
Yeah, that seems like a bug.  dnscontrol should have the same behavior as node.

I'd love a PR that fixes the problem (or at least one that documents the issue).  My guess is that the bug is somewhere in pkg/js/js.go and a few functions from https://golang.org/pkg/path/filepath and https://golang.org/pkg/os/#Chdir would be all you need to fix it.

Tom

**Comment by nlowe** [comment]
@captncraig That's what I was thinking as well. I'll try to get a PR up sometime this week. Since you're still 0.x how much to we care about breaking changes? Do I need to make the default behavior non-relative and add a flag for relative `require()`s?

**Comment by tlimoncelli** [comment]
I don't mind breaking behavior for require() as it wasn't documented either way.  We'll make a big note in the release notes to warn people.

(We're really at release 2.x not 0.2.x... I don't remember why I added a 0. when I set this all up.  We should fix that.)

**Comment by nlowe** [comment]
Fixed by #443
